src/validation/extensions/maxar/MaxarValidatorCommon.ts

```diff
@@ -0,0 +1,246 @@
+          referenceSystemPath,
+          "referenceSystem",
+          referenceSystem,
+          validReferenceSystemValues,
+          context
+        )
+      ) {
+        result = false;
+      }
+    }
+
+    // Validate epoch pattern
+    const epoch = srs.epoch;
+    if (defined(epoch)) {
+      const epochPath = path + "/epoch";
+      if (!BasicValidator.validateString(epochPath, "epoch", epoch, context)) {
+        result = false;
+      } else {
+        const epochPattern = /^[0-9]+(\.[0-9]+)?$/;
+        if (!epochPattern.test(epoch)) {
+          const message =
+            "The 'epoch' must match the pattern '^[0-9]+(\\.[0-9]+)?$'";
+          const issue = JsonValidationIssues.STRING_VALUE_INVALID(
+            epochPath,
+            message
+          );
+          context.addIssue(issue);
+          result = false;
+        }
+      }
+    }
+
+    // Validate coordinateSystem
+    const coordinateSystem = srs.coordinateSystem;
+    if (defined(coordinateSystem)) {
+      const coordinateSystemPath = path + "/coordinateSystem";
+      if (
+        !BasicValidator.validateString(
+          coordinateSystemPath,
+          "coordinateSystem",
+          coordinateSystem,
+          context
+        )
+      ) {
+        result = false;
+      } else {
+        if (
+          !MaxarValidatorCommon.validateCoordinateSystem(
+            coordinateSystemPath,
+            coordinateSystem,
+            context
+          )
+        ) {
+          result = false;
+        }
+      }
+    }
+
+    // Validate elevation enum
+    const elevation = srs.elevation;
+    if (defined(elevation)) {
+      const elevationPath = path + "/elevation";
+      const validElevationValues = ["ELLIPSOID", "EGM2008"];
+      if (
+        !BasicValidator.validateEnum(
+          elevationPath,
+          "elevation",
+          elevation,
+          validElevationValues,
+          context
+        )
+      ) {
+        result = false;
+      }
+    }
+
+    // Validate optional properties with defaults
+    result =
+      MaxarValidatorCommon.validateOptionalSrsProperties(path, srs, context) &&
+      result;
+
+    return result;
+  }
+
+  /**
+   * Validates the coordinateSystem property which has complex validation rules
+   */
+  static validateCoordinateSystem(
+    path: string,
+    coordinateSystem: string,
+    context: ValidationContext
+  ): boolean {
+    // Check if it's one of the simple enum values
+    const simpleValues = ["GEOD", "ECEF"];
+    if (simpleValues.includes(coordinateSystem)) {
+      return true;
+    }
+
+    // Check UTM pattern: UTM(01-60)[NS]
+    const utmPattern = /^UTM(0[123456789]|60|[12345][0123456789])[NS]$/;
+    if (utmPattern.test(coordinateSystem)) {
+      return true;
+    }
+
+    // Check S2 pattern: S2F[1-6]
+    const s2Pattern = /^S2F[123456]$/;
+    if (s2Pattern.test(coordinateSystem)) {
+      return true;
+    }
+
+    // If none of the patterns match, it's invalid
+    const message =
```

Comment:
Nitpick (?): I usually tried to be _very_ elaborate with the error messages. Specifically: I tried to _always_ include the actual value that was found. 

(A message like `"The value is wrong"` is far less helpful than `"The value must be X but was Y"`...)